jobs/admin.py

```diff
@@ -22,9 +33,75 @@ class JobAdmin(admin.ModelAdmin):
         models.TextField: {'widget': AutosizedTextarea},
     }
 
+    def get_urls(self):
+        urls = super(JobAdmin, self).get_urls()
+        my_urls = patterns('',
+            url(
+                r'^(?P<id>\d+)/assign/$',
+                self.assign_job_reviewer,
```

Comment:
Wrap each view with ```self.admin_site.admin_view()```; this means it will use the default admin permissions - see docs here: https://docs.djangoproject.com/en/1.7/ref/contrib/admin/
e..g
```
    url(
        r'^(?P<id>\d+)/assign/$',
        self.admin_site.admin_view(self.assign_job_reviewer),
        name='assign_job_reviewer'
    )
```